common/http/src/main/java/io/helidon/common/http/HttpMediaType.java

```diff
@@ -191,6 +193,19 @@ default HttpMediaType withCharset(String charset) {
                 .build();
     }
 
+    /**
+     * Create a new {@link io.helidon.common.http.HttpMediaType} instance with the same type, subtype and parameters
+     * copied from the original instance and the supplied {@value #CHARSET_PARAMETER} parameter.
+     *
+     * @param charset the {@value #CHARSET_PARAMETER} parameter value
+     * @return copy of the current {@code MediaType} instance with the {@value #CHARSET_PARAMETER}
+     *         parameter set to the supplied value.
+     */
+    default HttpMediaType withCharset(Charset charset) {
+        Objects.requireNonNull(charset);
+        return withCharset(charset.name());
```

Comment:
`charset.name()` will trigger a `NullPointerException` for you that is "smart"; no need for `Objects.requireNonNull(charset)`

Reply:
Removed the check.